File: nowa/2015/python/zad4_1.py
# ladowanie danych
with open('../dane/liczby.txt') as f:
    data = []
    for line in f.readlines():
        data.append(line[:-1])

# liczenie liczb z wieksza iloscia zer niz jedynek
more_zeros = 0
for line in data:
    zeros = 0
    ones = 0
    for char in line:
        if char == '1':
            ones += 1
        else:
            zeros += 1
    if zeros > ones:
        more_zeros += 1

# zapisanie odpowiedzi
answer = f'4.1. Liczby z wieksza iloscia zer: {more_zeros}'
print(answer)

File: nowa/2015/c++/zad4_1.cpp
#include <iostream>
#include <cstdlib>
#include <fstream>
#include <string>
#include <vector>
using namespace std;

string zad4_1()
{
    string line;
    vector<string> content;

    fstream file("../dane/liczby.txt");

    //wczytywanie danych z pliku
    if (file.is_open())
    {
        while (getline(file, line))
            content.push_back(line);
    }
    file.close();

    //liczenie linii, które mają więcej zer niż jedynek
    int howManyLines = 0, zeros = 0, ones = 0;

    for (int i = 0; i < content.size(); i++)
    {
        for (int j = 0; j < content[i].size(); j++)
        {
            if (content[i][j] == '0')
                zeros++;
            else
                ones++;
        }

        if (zeros > ones)
            howManyLines++;

        zeros = 0;
        ones = 0;
    }

    return "4.1. Liczba linii, w ktorych jest wiecej zer niz jedynek: " + to_string(howManyLines);
}